### ISSUE TITLE
Add Field Filters for Kanji Grid

### ISSUE THREAD

**Issue by HatOnAFox** [jyore/anki-kanji-grid#7]
**Functionality:**
Currently the Kanji Grid is configured to consider all fields for a card to measure "total exposure". 

**Problem:**
There are instances where people use pre-made decks where kanji that isn't being repped appears in a field like "Story". As a result when the card is repped the current Kanji Grid interprets these additional Kanji as having been "repped" when in fact they have not.

**Solution:**
In the GUI for the Kanji Grid an checkbox for "Pre-Made Deck" or "Ignore Deck Fields" should be added. If the box is checked two things should happen:

**1.** A drop down should be added where the user can input a field to be ignored
**2.** Underneath the dropdown a button to add another field should appear and the list of fields available to ignore should only show fields that have not been selected previously to be ignored.

**Alternative solution:** If possible make the fields to ignore a "Multi-Select" field so that the GUI is cleaner.

**NOTE:** A set of instructions should be included in the GUI so that people understand the new functionality and whether it is applicable prior to generating a grid. Additionally, adding a prompt after someone clicks to have the grid generated that they are "sure" they want to generate the grid with a warning about pre-made decks might help. This should have a checkbox in the prompt to prevent showing the warning in the future.

**Comment by jyore** [comment]
Currently in my deck, I have kanji in my audio title fields and it is picking it up as review, so this feature is 100% needed. As such, I have been thinking on implementation

I am wondering if I could get a submenu under each deck that allows you to check/uncheck which fields to use, directly underneath the deck. This way you can easily include fields based on deck and you are not avoiding all fields named a certain way, because of a string match.

Basically, it would look like additional checkboxes, indented a little bit, underneath each list item. I think it would be cool if the menu items were collapsable so you could hide/show the fields in the deck.

**Comment by jyore** [comment]
I have implemented this feature and it is in the master branch now.

I decided to take the idea of drop downs, but manage them at a card-field-model level. The settings UI has a new box on the right side that will populate with all of the note types that are in your collection. When you expand the menu for each note type, it will  give you a list of check boxes for each field that is part of the note type. If you uncheck a box, that field will not be scanned any longer. The menu is completely stateful, as well, so your settings will be remembered.

![2018_08_31_6_48](https://user-images.githubusercontent.com/904738/44914609-5f8fec00-acf6-11e8-947c-c0abe2b78aed.png)
